src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -687,12 +728,32 @@ private static bool IsArrayCompatibleReadOnlyInterface(Type type)
             return allProperties;
         }
 
+        [RequiresUnreferencedCode(PropertyTrimmingWarningMessage)]
+        private static object? BindParameter(ParameterInfo parameter, IConfiguration config, BinderOptions options)
+        {
+            string parameterName = parameter.Name!;
```

Comment:
Here's another case where trimming would fail here. There is an option with the trimmer that will remove parameter names. This is already marked `RequiresUnreferencedCode`. This is just another reason why it is marked not trim compatible.

A source generator for config binding shouldn't have this problem.

cc @vitek-karas

Reply:
I'm not entirely sure I understand the comment. I'll read up on this, but in the meantime, is there something that I need to change here? I copied that attribute from `BindProperty`; a [semi]-educated guess told me it was probably relevant and applicable.

Reply:
No, there isn't anything you can do here. This code is already marked as not trimming compatible.